sql_generators/fenix_use_counters/__init__.py

```diff
@@ -0,0 +1,109 @@
+import os
+import shutil
+from pathlib import Path
+
+import click
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+from bigquery_etl.cli.utils import use_cloud_function_option
+from bigquery_etl.format_sql.formatter import reformat
+from bigquery_etl.schema import SCHEMA_FILE, Schema
+from bigquery_etl.util.common import write_sql
+
+FILE_PATH = Path(os.path.dirname(__file__))
+BASE_DIR = Path(FILE_PATH).parent.parent
+TEMPLATE_CONFIG = FILE_PATH / "templates/templating.yaml"
+
+
+def generate_query(project, dataset, destination_table, write_dir):
+    """Generate feature usage table query."""
+    with open(TEMPLATE_CONFIG, "r") as f:
+        render_kwargs = yaml.safe_load(f) or {}
+    env = Environment(loader=FileSystemLoader(str(FILE_PATH / "templates")))
+    template = env.get_template("query.sql")
+
+    write_sql(
+        write_dir / project,
+        f"{project}.{dataset}.{destination_table}",
+        "query.sql",
+        reformat(template.render(**render_kwargs)),
+    )
+
+
+def generate_view(project, dataset, destination_table, write_dir):
+    """Generate feature usage table view."""
+    view_name = "fenix_use_counters_v2"
+    view_dataset = "fenix"
+
+    sql = reformat(
+        f"""
+        CREATE OR REPLACE VIEW `{project}.{view_dataset}.{view_name}` AS
+        SELECT
+            *
+        FROM
+            `{project}.{dataset}.{destination_table}`
+    """
+    )
+
+    write_sql(
+        write_dir / project, f"{project}.{view_dataset}.{view_name}", "view.sql", sql
+    )
+
+
+def generate_metadata(project, dataset, destination_table, write_dir):
+    """Copy metadata.yaml file to destination directory."""
+    shutil.copyfile(
+        FILE_PATH / "templates" / "metadata.yaml",
+        write_dir / project / dataset / destination_table / "metadata.yaml",
+    )
+
+def generate_schema(project, dataset, destination_table, write_dir):
+    """Generate the table schema."""
+    # get schema
+    table_schema = Schema.for_table(project, dataset, destination_table)
```

Comment:
I'm not following the use of the production schema here? Won't this `sql_generator` be the only thing changing it?

Reply:
So if you don't do this step, it won't actually make the table and it needs to make the table first  - the table actually stays the same format even as more columns get added to the template YAML file because the table is a pivoted format (where all the bajillion wide columns get pivoted into a single metric column)